tests/pipeline/test_events.py

```diff
@@ -0,0 +1,191 @@
+"""
+Tests for setting up an EventsLoader and a BlazeEventsLoader.
+"""
+from nose_parameterized import parameterized
+
+import blaze as bz
+import pandas as pd
+from pandas.util.testing import assert_series_equal, TestCase, assertRaises
+
+from zipline.pipeline.data import DataSet, Column
+from zipline.pipeline.loaders.blaze.events import BlazeEventsLoader
+from zipline.pipeline.loaders.events import (
+    BAD_DATA_FORMAT_ERROR,
+    DF_NO_TS_NOT_INFER_TS_ERROR,
+    DTINDEX_NOT_INFER_TS_ERROR,
+    EventsLoader,
+    SERIES_NO_DTINDEX_ERROR,
+    SID_FIELD_NAME,
+    TS_FIELD_NAME,
+    WRONG_COLS_ERROR,
+)
+from zipline.utils.memoize import lazyval
+from zipline.utils.numpy_utils import datetime64ns_dtype
+
+ABSTRACT_METHODS_ERROR = 'abstract methods concrete_loader'
+
+DAYS_SINCE_PREV = 'days_since_prev'
+
+PREVIOUS_ANNOUNCEMENT = 'previous_announcement'
+
+ANNOUNCEMENT_FIELD_NAME = 'announcement_date'
+
+
+class EventDataSet(DataSet):
+    previous_announcement = Column(datetime64ns_dtype)
+
+
+class EventDataSetLoader(EventsLoader):
+
+    def __init__(self,
+                 all_dates,
+                 events_by_sid,
+                 infer_timestamps=False,
+                 dataset=EventDataSet):
+        super(EventDataSetLoader, self).__init__(
+            all_dates,
+            events_by_sid,
+            infer_timestamps=infer_timestamps,
+            dataset=dataset,
+        )
+
+    @property
+    def expected_cols(self):
+        return frozenset([ANNOUNCEMENT_FIELD_NAME])
+
+    @lazyval
+    def previous_announcement_loader(self):
+        return self._previous_event_date_loader(
+            self.dataset.previous_announcement,
+            ANNOUNCEMENT_FIELD_NAME,
+        )
+
+    @lazyval
+    def next_announcement_loader(self):
+        return self._previous_event_date_loader(
+            self.dataset.previous_announcement,
+            ANNOUNCEMENT_FIELD_NAME,
+        )
+
+
+class EventDataSetLoaderNoExpectedCols(EventsLoader):
+
+    def __init__(self,
+                 all_dates,
+                 events_by_sid,
+                 infer_timestamps=False,
+                 dataset=EventDataSet):
+        super(EventDataSetLoaderNoExpectedCols, self).__init__(
+            all_dates,
+            events_by_sid,
+            infer_timestamps=infer_timestamps,
+            dataset=dataset,
+        )
+
+
+dtx = pd.date_range('2014-01-01', '2014-01-10')
+
+
+def assert_loader_error(events_by_sid, error, msg, infer_timestamps=True):
```

Comment:
idk if this should have a default arg since we should be explicit about this in the tests.